docker/config/config.go

```diff
@@ -55,15 +55,19 @@ func (c *Config) GetCredentials(registry string) (string, string, bool) {
 }
 
 func getAuthJSONString(username, password string) string {
-	if username == "_json_key" {
-		return fmt.Sprintf("%s:%s", username, password)
+	b, err := json.Marshal(types.AuthConfig{
```

Comment:
Considering using an anonymous struct here as well to reduce the dependencies, but I wanted to denote where the fields come from - it _may_ assist future contributors in discovering what authn knobs are available.